portfolio/src/main/webapp/script.js

```diff
@@ -20,7 +20,22 @@ function off() {
 }
 // asynchronously fetch content from server
 async function getContent() {
-  const response = await fetch('/data');
-  const quote = await response.text();
-  document.getElementById('content-container').innerText = quote;
+    const response = await fetch('/data');
+    const hardCodedContent = await response.json();
+  
+    const contentListElement = document.getElementById("content-container");
+    contentListElement.innerHTML = '';
+    contentListElement.appendChild(
+        createListElement("Element 0: " + hardCodedContent[0]));
+    contentListElement.appendChild(
+        createListElement("Element 1: " + hardCodedContent[1]));
+    contentListElement.appendChild(
+        createListElement("Element 2: " + hardCodedContent[2]));
```

Comment:
This is no longer hard coded content anymore as you updated this PR to also handle the post.  So you will need to update this to handle a variable number of comments.

Reply:
This part was not addressed in the part 4 tutorial. It seems like handling a variable set of data will go in hand with datastore for part 5 of walkthrough, so I will change that in the next pull request (part 5).

Reply:
I see the confusion.

Part 5 of the tutorial handles persistent storage.

For part 4 you still need to handle displaying a variable number of comments here.  The way you have it coded right now you are indexing out of the array when no comments are posted or when 1 comment is submitted.

Reply:
Oh I see, has been fixed!